Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.mm

```diff
@@ -0,0 +1,321 @@
+    SET_NSURLREQUESTDATA_PRIMITIVE(failInsecureLoadWithHTTPSDNSRecord, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(isWebSearchContent, NSNumber, bool);
+
+    NSArray *contentDispositionEncodingFallbackArray = dict[@"contentDispositionEncodingFallbackArray"];
+    if (contentDispositionEncodingFallbackArray != nil && [contentDispositionEncodingFallbackArray isKindOfClass:[NSArray class]]) {
+        Vector<CoreIPCNumber> vector;
+        vector.reserveInitialCapacity(contentDispositionEncodingFallbackArray.count);
+        for (id element in contentDispositionEncodingFallbackArray) {
+            if ([element isKindOfClass:[NSNumber class]]) {
+                CoreIPCNumber tooAdd(element);
+                vector.append(WTFMove(tooAdd));
+            }
+        }
+        vector.shrinkToFit();
+        m_data.contentDispositionEncodingFallbackArray = WTFMove(vector);
+    }
+}
+
+CoreIPCNSURLRequest::CoreIPCNSURLRequest(CoreIPCNSURLRequestData&& data)
+    : m_data(WTFMove(data)) { }
+
+CoreIPCNSURLRequest::CoreIPCNSURLRequest(const RetainPtr<NSURLRequest>& request)
+    : CoreIPCNSURLRequest(request.get()) { }
+
+
+RetainPtr<id> CoreIPCNSURLRequest::toID() const
+{
+    RetainPtr<NSMutableDictionary> dict = adoptNS([[NSMutableDictionary alloc] initWithCapacity:19]);
+
+    SET_DICT_FROM_OPTIONAL_MEMBER(protocolProperties);
+    SET_DICT_FROM_PRIMITIVE(isMutable, NSNumber, Bool);
+    SET_DICT_FROM_MEMBER(URL);
+    SET_DICT_FROM_PRIMITIVE(timeout, NSNumber, Double);
+
+    [dict setObject:[NSNumber numberWithUnsignedChar:static_cast<uint8_t>(m_data.cachePolicy)] forKey:@"cachePolicy"];
+
+    SET_DICT_FROM_OPTIONAL_MEMBER(mainDocumentURL);
+    SET_DICT_FROM_PRIMITIVE(shouldHandleHTTPCookies, NSNumber, Bool);
+
+    [dict setObject:[NSNumber numberWithShort:static_cast<int16_t>(m_data.explicitFlags.toRaw())] forKey:@"explicitFlags"];
+
+    SET_DICT_FROM_PRIMITIVE(allowCellular, NSNumber, Bool);
+    SET_DICT_FROM_PRIMITIVE(preventsIdleSystemSleep, NSNumber, Bool);
+    SET_DICT_FROM_PRIMITIVE(timeWindowDelay, NSNumber, Double);
+    SET_DICT_FROM_PRIMITIVE(timeWindowDuration, NSNumber, Double);
+
+    [dict setObject:[NSNumber numberWithUnsignedChar:static_cast<uint8_t>(m_data.networkServiceType)] forKey:@"networkServiceType"];
+
+    SET_DICT_FROM_PRIMITIVE(requestPriority, NSNumber, Int);
+    SET_DICT_FROM_OPTIONAL_PRIMITIVE(isHTTP, NSNumber, Bool);
+    SET_DICT_FROM_OPTIONAL_MEMBER(httpMethod);
+
+    if (m_data.headerFields) {
+        RetainPtr<NSMutableDictionary> headerFields = adoptNS([[NSMutableDictionary alloc] initWithCapacity:(*m_data.headerFields).size()]);
+        for (auto& headerPair : *m_data.headerFields) {
+            WTF::switchOn(headerPair.second,
+                [&] (const String s) {
+                    [headerFields setObject:(NSString *)s forKey:(NSString *)headerPair.first];
+                },
+                [&] (const Vector<String> a) {
```

Comment:
(const Vector<String>& vector)
Prevents unnecessary copies and has a better variable name.